Smooth camera drag.

cisar2218 opened the pull request:
- Camera drag separated from swipe: created `OnDragEvent` with `Vector3` position difference.
- Events in `PlayScreenState` reassigned to new `OnDragEvent`.

- Not tested on actual mobile device => drag speed value may need adjustments.
- In future drag speed could be based on zoom.
- Camera is not clambed to reasonable map radius yet.

cisar2218 committed:
Smooth camera drag.

- Camera drag separated from swipe: `OnDragEvent` with `Vector3` position difference.
- Events in `PlayScreenState` reassigned to new `OnDragEvent`

darkfriend77 reviewed (changes requested):
Nice, thx looks good.

I tested it, and found the following behavior, that broke on the PR, interaction with the board should only be possible on the Board screen, ray cast should be blocked by the overlapping UI, parts.

I added a picture to visualize it for a better understanding.

![image](https://github.com/SubstrateGaming/Substrate.Hexalem.Unity/assets/17710198/503619dc-f520-4736-b3e6-8420fe26c28f)

cisar2218 committed:
Begin movement of camera only if pointer not over UI.

cisar2218 committed:
Missed change in prev commit.

cisar2218 commented:
Thx for feedback. I added the check that should do the work. However problem is that pointer callback is sometimes resolved after input callback. Meaning the current UI detection is not 100% accurate (at least in my environment).

Current pointer callbacks:
```cs
_root.Q("BottomBound").RegisterCallback<PointerEnterEvent>(e => isPointerOverUI = true);
_root.Q("FloatBody").RegisterCallback<PointerEnterEvent>(e => isPointerOverUI = false);
```

Should I couple the checks to handle input function to remove the issue? Since it's used only for draging and swiping in the code currently. Like so:

```cs
case TouchPhase.Moved:
    touchEnd = position;
        if (!isSwiping && /* manual check with new function here: */ isPointerOverUI() && Vector2.Distance(touchStart, touchEnd) >= swipeThreshold)
        {
            ProcessSwipe(touchEnd.x - touchStart.x, touchEnd.y - touchStart.y);
            isSwiping = true;
        }
        break;
```

darkfriend77 commented:
Okay, I tested it. I did the following test.

When you switch from trying to swipe from the board screen to the selection screen and vice versa, the first attempt acts the opposite way as expected. It swipes on the Selection and it blocks on the Board, seems a logic issue. repeating attempts in the same space are then correctly handled.

Recreate the test.

```
- Try Swipe 3 x in Board
- Try Swipe 3 x in Selection
...
```

Hope that helps to fix the issue.
